lib/backend/k8s/resources/resources.go

```diff
@@ -100,3 +102,74 @@ func SetCalicoMetadataFromK8sAnnotations(calicoRes Resource, k8sRes Resource) {
 		}
 	}
 }
+
+// Store Calico Metadata in the in the k8s resource annotations for CRD backed resources.
+// This should store all Metadata except for those stored in Annotations and Labels and
+// store them in annotations.
+func StoreCalicoMetadataInK8sAnnotations(res Resource) error {
+	metadataBytes, err := json.Marshal(res.GetObjectMeta().(*metav1.ObjectMeta))
+	if err != nil {
+		return err
+	}
+	annotations := res.GetObjectMeta().GetAnnotations()
+	if len(annotations) == 0 {
+		annotations = make(map[string]string)
+	}
+	annotations[metadataAnnotation] = string(metadataBytes)
+	res.GetObjectMeta().SetAnnotations(annotations)
+
+	// Make sure to clear out all of the Calico Metadata out of the ObjectMeta except for
+	// Name, Namespace, and ResourceVersion.
+	res.GetObjectMeta().SetSelfLink("")
+	res.GetObjectMeta().SetUID("")
+	res.GetObjectMeta().SetGeneration(int64(0))
+	var emptyTime metav1.Time
+	res.GetObjectMeta().SetCreationTimestamp(emptyTime)
```

Comment:
I don't think you need these var lines.  You can just do:  `res.GetObjectMeta().SetCreationTimestamp(metav1.Time{})`

same for the others below

Reply:
I tried using `metav1.Time{}` but it gave me an empty Time object instead of the zero value for a Time object.  Ideally I'm hoping to remove the `CreationTimestamp` and other fields as if they didn't exist instead of filling in the values with empty ones. I did a poor job naming the variables there though so I can fix that.

Reply:
Just retested it. Must've gotten it confused with the slices. `metav1.Time{}` is the empty/zero value.  I'll make that change as well.